tools/godoctests/pkg/analyzer/analyzer_test.go

```diff
@@ -0,0 +1,18 @@
+package analyzer
+
+import (
+	"os"
+	"path/filepath"
+	"testing"
+
+	"golang.org/x/tools/go/analysis/analysistest"
+)
+
+// TestAnalyzer runs the analyzer on the test functions in testdata/funcs.go
+func TestAnalyzer(t *testing.T) {
+	f, err := os.Getwd()
+	if err != nil {
+		t.Fatal("failed to get working directory", err)
+	}
+	analysistest.Run(t, filepath.Join(f, "testdata"), Analyzer, ".")
```

Comment:
I feel like something's missing.  Where do we declare what the expected results of this test are?  Oh, I see, it's the "// want" comments.  Might be worth adding a note here, since this is kind of non-obvious.